### PR TITLE
[Java] Rename DinoComparer to StringComparer and update implementation

### PR DESCRIPTION
**Pull request by Aliandi**
# Description
- Change DinoComparer to StringComparer for more clarity within the code. This method will take two strings and compare them, taking into consideration that they may be null or empty. 
- The compare method was refactored for a more clear implementation.
- The dependant class NumberWithUnitExtractor was updated to work with the name change.

All the tests passed succesfully

imagen

### CODE REVIEW

**Comment by denscollo** [comment]
LGTM

File: Java/libraries/recognizers-text-number-with-unit/src/main/java/com/microsoft/recognizers/text/numberwithunit/utilities/StringComparer.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package com.microsoft.recognizers.text.numberwithunit.utilities;
+
+import java.util.Comparator;
+
+public class StringComparer implements Comparator<String> {
+    @Override
+    public int compare(String stringA, String stringB) {
+        if (stringA.isEmpty() && stringB.isEmpty()) {
```
Comment (JuanAr) [inline Java/libraries/recognizers-text-number-with-unit/src/main/java/com/microsoft/recognizers/text/numberwithunit/utilities/StringComparer.java]: Does this work well if `stringA` or `stringB` is `null`?

Comment (JuanAr) [reply Java/libraries/recognizers-text-number-with-unit/src/main/java/com/microsoft/recognizers/text/numberwithunit/utilities/StringComparer.java]: https://stackoverflow.com/a/8970033

**Review by gasper-az — APPROVED** [review]

**Review by JuanAr — APPROVED** [review]
👍

**Review by ParadoxARG — APPROVED** [review]
:shipit:

**Comment by gasper-az** [comment]
Closed due to MS#1422

**Pull request closed without merging**